### DISCUSSION TITLE
How to setup gitlab auth

### DISCUSSION THREAD

**Discussion by fspindle** [Kitware/CDash#2032] [Q&A] [answered]
After a successful installation of the server using docker with the last version from master branch (commit c2c430552fd), I'm unable to setup gitlab auth.

In my `.env` file I set
```
GITLAB_ENABLE=true
GITLAB_CLIENT_ID=de1bfc...
GITLAB_CLIENT_SECRET=gloas-...
GITLAB_DOMAIN=https://gitlab.com/
```

and I configure `gitlab.com` to create a user-owned application selecting `api`, `read_user`, `openid`, `profile`, `email` OAuth 2 Scopes. In the redirect URI I set `https://my-cdash.server.fr/auth/gitlab/callback`

then I run `docker compose --env-file .env ...`

But when I click on the Gitlab logo I'm redirected to a page with the following error 
```
An error has occurred
The redirect URI included is not valid.
```

I'm wondering what I'm doing wrong. Is my redirect URI address the right one ?

**Comment by josephsnyder** [comment, marked as answer]
@fspindle, thanks for the report.  I can replicate what you're experiencing locally.  

It seems like the fix is to specify the redirect URI in the configuration too.  When I added an `.env` entry for `GITLAB_REDIRECT_URI` which matched the value that I set when registering the user-owned application, I didn't receive the error any more.  Your block of `.env` would look like this:

```
GITLAB_ENABLE=true
GITLAB_CLIENT_ID=de1bfc...
GITLAB_CLIENT_SECRET=gloas-...
GITLAB_DOMAIN=https://gitlab.com/
GITLAB_REDIRECT_URI=https://my-cdash.server.fr/auth/gitlab/callback
```
Please give that a try and let me know if you can log in.  I'll do some digging on my end to determine why Socialite needs the `redirect_uri` in the `authorize` call when I would expect GitLab's configured value to handle it.

**Reply by josephsnyder** [reply]
And just for completeness sake: yes.  The redirect URL that you've set up is the correct one.

**Reply by fspindle** [reply]
@josephsnyder Thanks for your quick post. It works :-)
